tox.ini

```diff
@@ -38,32 +38,31 @@ commands =
 
 [docs]
 build_dir = _build
-# -W: make warnings into errors
-# --keep-going: continue on errors
-# -j: run with multiple processes
-sphinx_args = -W --keep-going -j auto
-# -v: verbose logging
-# -E: force rebuild of environment
-# -T: print traceback on error
-# -a: read/parse all files
-# -d: use tox's temp dir for caching
-sphinx_args_extra = {[docs]sphinx_args} -v -E -T -a -d {envtmpdir}/doctrees
+# replace when Sphinx 7.3 is released:
+#  -T -> --show-traceback
+#  -W -> --fail-on-warning
+#  -b -> --builder
+#  -v -> --verbose
+#  -a -> --write-all
+#  -E -> --fresh-env
+sphinx_args = -T -W --keep-going --jobs auto
```

Comment:
I decided to re-work all of this from my original implementation.

Using `-E`, `-a`, and `-d {envtmpdir}/doctrees` all the time makes all of this really slow since Sphinx basically needs to start from scratch every time. In your local environment, though, using all the caching really speeds things up. Use `tox -e docs -- -aE` if you want to start from scratch locally; `tox -e docs-all` will continue always re-reading everything, though.

Using `-v` in a terminal creates a worse experience since it needlessly lists all the files being read instead of using the screen redrawing to list the files being processed.